tracee-rules/engine/engine_test.go

```diff
@@ -0,0 +1,250 @@
+package engine
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeSignature struct {
+	getMetadata       func() (types.SignatureMetadata, error)
+	getSelectedEvents func() ([]types.SignatureEventSelector, error)
+	init              func(types.SignatureHandler) error
+	onEvent           func(types.Event) error
+	onSignal          func(signal types.Signal) error
+}
+
+func (fs fakeSignature) GetMetadata() (types.SignatureMetadata, error) {
+	if fs.getMetadata != nil {
+		return fs.getMetadata()
+	}
+
+	return types.SignatureMetadata{
+		Name: "Fake Signature",
+	}, nil
+}
+
+func (fs fakeSignature) GetSelectedEvents() ([]types.SignatureEventSelector, error) {
+	if fs.getSelectedEvents != nil {
+		return fs.getSelectedEvents()
+	}
+
+	return []types.SignatureEventSelector{}, nil
+}
+
+func (fs fakeSignature) Init(cb types.SignatureHandler) error {
+	if fs.init != nil {
+		return fs.init(cb)
+	}
+	return nil
+}
+
+func (fs fakeSignature) OnEvent(event types.Event) error {
+	if fs.onEvent != nil {
+		return fs.onEvent(event)
+	}
+	return nil
+}
+
+func (fs fakeSignature) OnSignal(signal types.Signal) error {
+	if fs.onSignal != nil {
+		return fs.onSignal(signal)
+	}
+	return nil
+}
+
+func TestEngine_consumeSource(t *testing.T) {
+	testCases := []struct {
+		name              string
+		inputEvent        types.TraceeEvent
+		inputSignature    fakeSignature
+		expectedNumEvents int
+		expectedError     string
+	}{
+		{
+			name: "happy path - with one matching selector",
+			inputEvent: types.TraceeEvent{
+				EventName: "test_event",
+				ProcessID: 2, ParentProcessID: 1,
+				Args: []types.TraceeEventArgument{{
+					Name: "test_foo",
```

Comment:
it isn't realistic for an arg to have no value. even though it's not enforced better to put something here or remove the entire arg (it's not used is it). same in all other place

Reply:
Yep - this is cruft and should be removed.